### PR TITLE
Don't download inputdata for files with accessURL

### LINKED ISSUE #1698: LHCbDataset downloads data for local jobs
When running a simple DaVinci job locally with `LHCbDataset` for the inputdata it downloads the file. It doesn't seem to be an issue for `LHCbCompressedDataset` though.

### PR DESCRIPTION
**Pull request by mesmith75**
Fixes #1698 .

This checks if a file from the inputdata has an accessURL . If it does then the inputdata won't be downloaded.

### CODE REVIEW

File: ganga/GangaCore/GPIDev/Lib/File/OutputFileManager.py [hunk 1]
```diff
@@ -230,18 +230,34 @@ def doIHaveInputFiles(job):
...
     if job.inputdata:
         if job.inputdata and isType(job.inputdata, GangaDataset):
-            inputfiles_list += job.inputdata.files
+            for _f in job.inputdata:
+                try:
+                    _url = _f.accessURL()
+                    if _url:
+                        continue
+                    else:
+                        inputfiles_list.append(_f)
```
Comment (egede) [inline ganga/GangaCore/GPIDev/Lib/File/OutputFileManager.py]:
```python
if not _f.accessURL():
    inputfiles_list.append(_f)
```
seems shorter and clearer

File: ganga/GangaCore/GPIDev/Lib/File/OutputFileManager.py [hunk 2]
```diff
@@ -230,18 +230,34 @@ def doIHaveInputFiles(job):
...
     elif job.master is not None:
         if job.master.inputdata and isType(job.master.inputdata, GangaDataset):
-            inputfiles_list += job.master.inputdata.files
+            for _f in job.master.inputdata:
+                try:
+                    _url = _f.accessURL()
+                    if _url:
+                        continue
+                    else:
+                        inputfiles_list.append(_f)
```
Comment (egede) [inline ganga/GangaCore/GPIDev/Lib/File/OutputFileManager.py]: as above

**Review by egede — CHANGES_REQUESTED** [review]
I am a bit confused about what problem you are solving here. Is it about downloading to the worker node or about downloading after the job has finished?

**Comment by mesmith75** [comment]
This solves #1698 . When running local or interactive jobs ganga is downloading remote inputdata to the WN. It doesn't make sense because the `data.py` file we create uses the remote access so we are filling up the tmp directory with massive data files for no reason.

**Review by egede — APPROVED** [review]

**Pull request merged**